tensordict/utils.py

```diff
@@ -1769,27 +1769,29 @@ def _check_keys(
     )
     # TODO: compile doesn't like set() over an arbitrary object
     if is_compiling():
-        keys = {k for k in keys}  # noqa: C416
+        keys_set = {k for k in keys}  # noqa: C416
     else:
-        keys: set[str] = set(keys)
+        keys_set: set[str] = set(keys)
     for td in list_of_tensordicts[1:]:
         k = td.keys(
             include_nested=include_nested,
             leaves_only=leaves_only,
             is_leaf=_is_leaf_nontensor,
         )
         if not strict:
-            keys = keys.intersection(k)
+            keys_set = keys_set.intersection(k)
         else:
             if is_compiling():
                 k = {v for v in k}  # noqa: C416
             else:
                 k = set(k)
-            if k != keys:
+            if k != keys_set:
                 raise KeyError(
                     f"got keys {keys} and {set(td.keys())} which are incompatible"
                 )
-    return keys
+    if strict:
+        return keys
```

Comment:
```suggestion
        return list(keys)
```
pretty sure that's what you mean with your comment, but just to be on the safe side. Rn, the return type is not consistent with typing.